### ISSUE TITLE
Error [ERR_STREAM_DESTROYED]: Cannot call write after a stream was destroyed

### ISSUE THREAD

**Issue by filoucrackeur** [patrickjuchli/basic-ftp#165]
**Describe the bug**

When I top brutaly the ftp service on server side like vsftpd or switch off server during upload on download of file

**Example code**
I use standard code of the example trying to catch some exception but nothing to do

**Console output**
```
events.js:174
       throw er; // Unhandled 'error' event
       ^
 Error [ERR_STREAM_DESTROYED]: Cannot call write after a stream was destroyed
     at doWrite (_stream_writable.js:411:19)
     at writeOrBuffer (_stream_writable.js:399:5)
     at Socket.Writable.write (_stream_writable.js:299:11)
     at ReadStream.ondata (_stream_readable.js:710:20)
     at ReadStream.emit (events.js:198:13)
     at addChunk (_stream_readable.js:288:12)
     at readableAddChunk (_stream_readable.js:269:11)
     at ReadStream.Readable.push (_stream_readable.js:224:10)
     at lazyFs.read (internal/fs/streams.js:181:12)
     at FSReqWrap.wrapper [as oncomplete] (fs.js:467:17)
 Emitted 'error' event at:
     at errorOrDestroy (internal/streams/destroy.js:107:12)
     at onwriteError (_stream_writable.js:430:5)
     at onwrite (_stream_writable.js:461:5)
     at doWrite (_stream_writable.js:411:11)
     at writeOrBuffer (_stream_writable.js:399:5)
     [... lines matching original stack trace ...]
     at readableAddChunk (_stream_readable.js:269:11)
```
**Which version of Node.js are you using?**
node -v
```
v10.19.0
```
**Additional context**
Stop brutaly ftp service server side like vsftpd or switch off server

**Comment by patrickjuchli** [comment]
Do you see the same behaviour with the latest version 5.0.0?

**Comment by patrickjuchli** [comment]
This should be addressed at >= 5.0.0, do reopen if issue persists.
